spack-repo/packages/singularity-eos/package.py

```diff
@@ -5,7 +5,25 @@
 
 import os
 
+from spack.error import SpackError
 from spack.package import *
+from spack.directives import directive
+
+
+@directive("singularity_eos_plugins")
+def singularity_eos_plugin(name, path):
+    def _execute_register(pkg):
+        pkg.plugins[name] = path
+
+    return _execute_register
+
+
+def plugin_validator(pkg_name, variant_name, values):
+    if values == ("none",):
+        return
+    for v in values:
+        if v not in SingularityEos.plugins:
+            raise SpackError(f"Unknown Singularity-EOS plugin '{v}'")
```

Comment:
Should we check that `none` is not included in the list along with an actual plugin? Or does spack doe that validation for us?

Reply:
If someone writes `plugins=a,b,none` they will be yelled at by a Spack error:
```
Unknown Singularity-EOS plugin 'none'
```
Isn't that enough?

Reply:
Oh I see--- I thought that first if statement checked if `none` was contained in the list... but it's actually a check if it is the entire list. All is well. :+1: